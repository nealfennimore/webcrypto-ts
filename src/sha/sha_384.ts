import * as alg from "../alg";
import { WebCrypto } from "../crypto";
import { ShaShared } from "./shared";

export namespace SHA_384 {
    export const digest = (data: BufferSource) =>
        WebCrypto.digest(alg.SHA.Variant.SHA_384, data);

    export const hexify = ShaShared.hexify;
}
