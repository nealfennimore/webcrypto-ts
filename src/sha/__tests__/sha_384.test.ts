import { SHA_384 } from "../sha_384";

describe("SHA_384", () => {
    it("should hash", async () => {
        const data = encode("a message");
        const hash = await SHA_384.digest(data);
        expect(hash.byteLength).toEqual(48);
    });
    it("should hexify", async () => {
        const data = encode("a message");
        const hash = await SHA_384.digest(data);
        expect(SHA_384.hexify(hash)).toMatchSnapshot();
    });
});
