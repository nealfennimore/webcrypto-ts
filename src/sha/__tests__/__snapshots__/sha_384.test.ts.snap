// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHA_384 should hexify 1`] = `"9772402990d5de649a3e8acdbbb3120d3d9e254e21c970a6d2eb6fb2f29cf4940df0357553241ff5fcf56b277c1a9c92"`;
